rust/src/composepost.rs

```diff
@@ -254,6 +254,29 @@ fn postprocess_subs_dist(rootfs_dfd: &Dir) -> Result<()> {
     Ok(())
 }
 
+#[context("Cleaning up rpmdb leftovers")]
+pub(crate) fn postprocess_cleanup_rpmdb_impl(rootfs_dfd: &Dir) -> Result<()> {
+    let d = if let Some(d) = rootfs_dfd.open_dir_optional(RPMOSTREE_RPMDB_LOCATION)? {
+        d
+    } else {
+        return Ok(());
```

Comment:
Might be my non-Rust eyes getting confused by the `let d = if`, but would it be cleaner to move the `for` block into the `if`, drop the `else`, and have a single `Ok(())`?